ext/pdo_odbc/odbc_stmt.c

```diff
@@ -136,7 +136,11 @@ static int odbc_stmt_dtor(pdo_stmt_t *stmt)
 {
 	pdo_odbc_stmt *S = (pdo_odbc_stmt*)stmt->driver_data;
 
-	if (S->stmt != SQL_NULL_HANDLE) {
+	// TODO: Factor this out; pg/mysql/firebird do the same thing
+	bool server_obj_usable = !Z_ISUNDEF(stmt->database_object_handle)
+		&& IS_OBJ_VALID(EG(objects_store).object_buckets[Z_OBJ_HANDLE(stmt->database_object_handle)])
+		&& !(OBJ_FLAGS(Z_OBJ(stmt->database_object_handle)) & IS_OBJ_FREE_CALLED);
```

Comment:
This is super ugly. Wouldn't it be cleaner if the dbh would keep a list of stmts, so it could null stmt->H when being destroyed?

Reply:
That's more convoluted and consumes more memory, and the dbh destruction already destroys the statements too. Imo, this is good enough especially when hidden behind a function.

Reply:
Fair enough!

Could it make sense to move that into Zend/ to have a generic function? I think that not only PDO is affected by this; likely also other DB extensions, and even others (enchant broker/dictinonary comes to mind)?

Reply:
enchant shouldn't be affected as it does not create strong cycles and even does its own refcounting of the broker IIRC.
Other DB extension: could be affected indeed.